### DISCUSSION TITLE
Does 'queryClient.clear()' not valid with 'staleTime: Infinity'?

### DISCUSSION THREAD

**Discussion by vaynevayne** [TanStack/query#5206] [Q&A] [answered]
I have configured useQuery with staleTime infinity. When I log out and log back in, the page still uses the previous data and its isFetching seems to be false.

 "@tanstack/react-query": "^4.19.1",

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

It should clear all of the connected caches and therefore remove all query data from the cache. 

Please could you provide a minimal reproduction so I can take a look?
